tests/unit/src/test/scala/tests/CompilersLspSuite.scala

```diff
@@ -53,4 +56,284 @@ class CompilersLspSuite extends BaseCompletionLspSuite("compilers") {
+            s"""|${CreateNewSymbol.title("UniqueObject")}
+                |""".stripMargin,
+            kind = Nil,
+          )
+    } yield ()
+  }
+
+  test("never-compiling") {
+    cleanWorkspace()
+    for {
+      _ <- initialize(
+        """/metals.json
+          |{
+          |  "a": {}
+          |}
+          |/a/src/main/scala/a/A.scala
+          |package a
+          |class A {
+          |  // @@
+          |  def completeThisUniqueName(): String = 42
+          |}
+          |/a/src/main/scala/b/B.scala
+          |package b
+          |object UniqueObject {
+          |  def completeThisUniqueName() = 42
+          |  def completeThisUniqueName2(): String = 42
+          |}
+          |""".stripMargin
+      )
+      _ <- server.didOpen("a/src/main/scala/a/A.scala")
+      _ <- server.didOpen("a/src/main/scala/b/B.scala")
+      _ = assertNoDiff(
+        server.client.workspaceDiagnostics,
+        """|a/src/main/scala/a/A.scala:4:42: error: type mismatch;
+           | found   : Int(42)
+           | required: String
+           |  def completeThisUniqueName(): String = 42
+           |                                         ^^
+           |a/src/main/scala/b/B.scala:4:43: error: type mismatch;
+           | found   : Int(42)
+           | required: String
+           |  def completeThisUniqueName2(): String = 42
+           |                                          ^^
+           |""".stripMargin,
+      )
+      _ <- assertCompletion(
+        "b.UniqueObject.completeThisUniqueNa@@",
+        """|completeThisUniqueName(): Int
+           |completeThisUniqueName2(): String""".stripMargin,
+      )
+      _ <- assertCompletionEdit(
+        "UniqueObject@@",
+        """|package a
+           |
+           |import b.UniqueObject
+           |class A {
+           |  UniqueObject
+           |  def completeThisUniqueName(): String = 42
+           |}
+           |""".stripMargin,
+      )
+    } yield ()
+  }
+
+  test("never-compiling-reverse-order") {
+    cleanWorkspace()
+    for {
+      _ <- initialize(
+        """/metals.json
+          |{
+          |  "a": {}
+          |}
+          |/a/src/main/scala/a/A.scala
+          |package a
+          |class A {
+          |  // @@
+          |  def completeThisUniqueName(): String = 42
+          |}
+          |/a/src/main/scala/b/B.scala
+          |package b
+          |trait BTrait {
+          |  def completeThisUniqueName3() = 42
+          |}
+          |class B
+          |/a/src/main/scala/c/C.scala
+          |package c
+          |import b.BTrait
+          |import b.B
+          |
+          |object UniqueObject extends BTrait {
+          |  def completeThisUniqueName() = 42
+          |  def completeThisUniqueName2(b: B): String = 42
+          |}
+          |""".stripMargin
+      )
+      _ <- server.didOpen("a/src/main/scala/a/A.scala")
+      _ <- server.didOpen("a/src/main/scala/b/B.scala")
```

Comment:
This should work even if we don't open all the files, right?

Reply:
yes